### PR TITLE
1249 allure screenshot

### PR DESCRIPTION
**Pull request by ElenaPoduzova**
[#1249](https://github.com/jdi-testing/jdi-light/issues/1249) - Attach screenshots to Allure

### CODE REVIEW

File: jdi-light/src/main/java/com/epam/jdi/light/logger/JDILogger.java [hunk 1]
```diff
@@ -148,6 +180,10 @@ public void info(String s, Object... args) {
     }
     public void error(String s, Object... args) {
         logger.error(jdiMarker, getRecord(s, args));
+        writeToAllure(getRecord(s, args), FAILED);
+        if (screenshotStrategy.equals("on fail")) {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/logger/JDILogger.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4558480)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4558480&bid=15331575)

**Comment by ElenaPoduzova** [comment]
Need some updates

**Comment by ElenaPoduzova** [comment]
Added default screenshot.strategy value = "on fail". Add sample tests.

File: jdi-light/src/main/java/com/epam/jdi/light/logger/JDILogger.java [hunk 2]
```diff
@@ -116,18 +133,33 @@ public String getName() {
...
 
-    private static void writeToAllure(String message) {
+    private static void writeToAllure(String message, Status status) {
         if (!writeToAllure) return;
         final String uuid = UUID.randomUUID().toString();
-        StepResult step = new StepResult().withName(message).withStatus(PASSED);
+        StepResult step = new StepResult().withName(message).withStatus(status);
         getLifecycle().startStep(uuid, step);
+        if (FAILED.equals(status)) {
+            if ("on fail".equals(screenshotStrategy)) {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/logger/JDILogger.java]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4558480)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- jdi-light-examples/src/test/java/io/github/epam/tests/allurereport/AllureScreenshotTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4558480&bid=15331575)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light-examples/src/test/java/io/github/epam/tests/allurereport/AllureScreenshotTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4558480&bid=15331575)

File: jdi-light-examples/src/test/resources/allureScreenshot.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+<suite name="Test suite" parallel="methods" thread-count="1">
+    <test name="Tests">
+        <classes>
+            <class name="io.github.epam.tests.allurereport.AllureScreenshotTests"/>
+        </classes>
+    </test>
+    <listeners>
+        <listener class-name="io.github.epam.testng.TestNGListener" />
+    </listeners>
+</suite>
```
Comment (splekhanov) [inline jdi-light-examples/src/test/resources/allureScreenshot.xml]: Lets leave a new line in the end of the file

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light-examples/src/test/java/io/github/epam/tests/allurereport/AllureScreenshotTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4558480&bid=15331575)

**Pull request closed without merging**